### DISCUSSION TITLE
devtools typing issue for an edge case

### DISCUSSION THREAD

**Discussion by dai-shi** [pmndrs/zustand#2700] [Bug report]
### Bug Description

This causes an type error:
```ts
export const useStoreB = create<TableStore | null>()(
  devtools(() => null, { name: "table-storage" }),
);
```

It's rarely used like this, but it's unexpected. Can anyone dig into it?

### Reproduction Link

https://tsplay.dev/wRVZ5w

**Comment by AndyRightNow** [comment]
Hi @dai-shi , I made an attempt to fix the issue in this PR https://github.com/pmndrs/zustand/pull/2875 . Please kindly take a look and feel free to raise any review requests, thanks!

**Reply by dai-shi** [reply]
Thanks! I'll have a look later.
